plugin/src/main/java/com/h3xstream/findsecbugs/taintanalysis/TaintFrameModelingVisitor.java

```diff
@@ -539,30 +540,51 @@ private TaintMethodConfig getMethodConfig(InvokeInstruction obj) {
         return null;
     }
 
-    private TaintMethodConfig getConfigWithReplaceTags(
-            TaintMethodConfig config, String className, String methodName) {
-        if (!"java/lang/String".equals(className)) {
+    private TaintMethodConfig getConfigWithReplaceTags(TaintMethodConfig config, String className, String methodName) {
+
+        ObjectConfiguration objectConfiguration = new ObjectConfiguration(className, methodName);
+
+        if (!objectConfiguration.isAClassThatCanReplaceString()) {
+            return config;
+        }
+
+        /*
+          When Kotlin compiles the String replace method (when using regex values) Kotlin creates several instructions in its place.
```

Comment:
This issue was a bit complicated. When Kotlin compiles instructions like `tainted.replace("[\r\n]+".toRegex(), "")` it generates many instructions in its place. One of the issues is that when using Regex values, the `replace` method being called is from the `Regex` class rather the one in `StringsKt` class.

I didn't manage to find a way to extract the value of the regex at the time the `Regex#replace` method was called, so I had to store some state when the `Regex#<init>` method is being called and use it when `Regex#replace` method was used.

I don't like this solution, but it works. More than happy to change it in case you know of a better solution (particularly one where store state is not necessary).

Reply:
Good analysis. I will look in depth at your fix later.